src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+namespace SarifViewer
+{
+    public class ErrorListService
+    {
+        public static ErrorListService Instance = new ErrorListService();
+
+        public static void ProcessSarifLog(ResultLog resultLog)
+        {
+            foreach (RunLog runLog in resultLog.RunLogs)
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
```

Comment:
why is that such as help? with LINQ you can also easily write obscure or non-performant code. now, if you propose a LINQ pattern that is equally readable to foreach, and more concise, and no less efficient, I am interested.